test/test_python_type_stubs.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import textwrap
+from typing import List
+
+import unittest
+from mock import Mock
+
+from stone.api import Api, ApiNamespace
+from stone.data_type import (
+    Struct,
+    StructField,
+    Boolean,
+    List as StoneList,
+    UInt64,
+    Union,
+    UnionField,
+    Void,
+)
+from stone.target.python_type_stubs import PythonTypeStubsGenerator
+
+class TestPythonTypeStubsHelpers(object):
+    def _make_generator(self):
+        # type: () -> PythonTypeStubsGenerator
+        generator = PythonTypeStubsGenerator(
+            target_folder_path=Mock(),
+            args=Mock()
+        )
+        return generator
+
+    def _mock_emit(self, generator):
+        # type: (PythonTypeStubsGenerator) -> List[str]
+        """
+        Mock out PythonTypeStubsGenerator's .emit function, and return a list containing all params
+        emit was called with.
+        """
+        recorded_emits = []  # type: List[str]
+
+        def record_emit(str):
+            # type: (str) -> None
+            recorded_emits.append(str)
+
+        generator.emit_raw = Mock(side_effect=record_emit)  # type: ignore
+
+        return recorded_emits
+
+    def _make_namespace_with_two_structs(self):
+        ns = ApiNamespace('ns_with_two_structs')
+
+        a1 = Struct(name='A1', namespace=ns, token=None)
+        a1.set_attributes(None, [StructField('f1', Boolean(), None, None)])
+        ns.add_data_type(a1)
+
+        a2 = Struct(name='A2', namespace=ns, token=None)
+        a2.set_attributes(None, [StructField('f2', StoneList(UInt64()), None, None)])
+        ns.add_data_type(a2)
+
+        return ns
+
+    def _make_namespace_with_a_union(self):
+        ns = ApiNamespace('ns_with_a_union')
+
+        u1 = Union(name='U1', namespace=ns, token=None, closed=True)
+        u1.set_attributes(
+            doc=None,
+            fields=[
+                UnionField(name="first",
+                              doc=None,
+                              data_type=Void(),
+                              token=None
+                              ),
+                UnionField(name="last",
+                           doc=None,
+                           data_type=Void(),
+                           token=None
+                           ),
+            ],
+        )
+        ns.add_data_type(u1)
+
+        return ns
+
+    def _api(self):
+        api = Api(version="1.0")
+        return api
+
+class TestPythonTypeStubs(unittest.TestCase, TestPythonTypeStubsHelpers):
+    def test__generate_base_namespace_module__generates_stuff(self):
```

Comment:
Personal preference - puts a separator between <what's being tested> and <what's the expected behavior>, makes it easier to parse as a human IMO

(the equivalent of something like an RSpec like
```
describe "generate_base_namespace_module" do
  it "generates stuff" do
    test goes here
  end
end
```
)